### ISSUE TITLE
Gitea container hanging on SSH session - ARM32 image

### ISSUE THREAD

**Issue by mkenne11** [Kunde21/raspberry-dev#6]
Thanks for all your work maintaining an arm32 version of the Gitea image - it's muchly appreciated :)

Since the release of Gitea v1.12 the arm32 versions (both armv6 and armv7) haven't been working correctly for me. The container has been hanging when I send SSH requests to it. Btw it was working correctly for the v1.11.8 images.

I have compared the SSH session logs between the v1.11.8 and v1.12.3 image but haven't been able to track down the issue. However, below are snippets of the SSH service logs when I attempt to initiate a SSH session with my Gitea instance using the command:
```
ssh -T git@192.168.10.30 -vvv -p 2222
```

1. SSH session log for Gitea v1.11.8 (session connection successful)
```
...
debug1: Offering public key: /home/pi/.ssh/gitea-key RSA SHA256:pxkPu96w0CfEom3kXsSGjAUFrgY6pxPYlaOMVB+zBt8 explicit
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 60
debug1: Server accepts key: /home/pi/.ssh/gitea-key RSA SHA256:pxkPu96w0CfEom3kXsSGjAUFrgY6pxPYlaOMVB+zBt8 explicit
debug3: sign_and_send_pubkey: RSA SHA256:pxkPu96w0CfEom3kXsSGjAUFrgY6pxPYlaOMVB+zBt8
debug3: sign_and_send_pubkey: signing using rsa-sha2-512
debug3: send packet: type 50
debug3: receive packet: type 52
debug1: Authentication succeeded (publickey).
Authenticated to 192.168.10.30 ([192.168.10.30]:2222).
debug1: channel 0: new [client-session]
debug3: ssh_session2_open: channel_new: 0
debug2: channel 0: send open
debug3: send packet: type 90
debug1: Requesting no-more-sessions@openssh.com
debug3: send packet: type 80
debug1: Entering interactive session.
debug1: pledge: network
debug3: receive packet: type 80
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug3: receive packet: type 4
debug1: Remote: /data/git/.ssh/authorized_keys:2: key options: command user-rc
debug3: receive packet: type 4
debug1: Remote: /data/git/.ssh/authorized_keys:2: key options: command user-rc
debug3: receive packet: type 91
debug2: channel_input_open_confirmation: channel 0: callback start
debug2: fd 3 setting TCP_NODELAY
debug3: ssh_packet_set_tos: set IP_TOS 0x08
debug2: client_session2_setup: id 0
debug1: Sending environment.
debug3: Ignored env SHELL
debug3: Ignored env PWD
debug3: Ignored env LOGNAME
debug3: Ignored env XDG_SESSION_TYPE
debug3: Ignored env HOME
debug1: Sending env LANG = en_GB.UTF-8
debug2: channel 0: request env confirm 0
debug3: send packet: type 98
debug3: Ignored env LS_COLORS
debug3: Ignored env SSH_CONNECTION
debug3: Ignored env XDG_SESSION_CLASS
debug3: Ignored env TERM
debug3: Ignored env USER
debug3: Ignored env SHLVL
debug3: Ignored env XDG_SESSION_ID
debug3: Ignored env XDG_RUNTIME_DIR
debug3: Ignored env SSH_CLIENT
debug3: Ignored env PATH
debug3: Ignored env DBUS_SESSION_BUS_ADDRESS
debug3: Ignored env MAIL
debug3: Ignored env SSH_TTY
debug3: Ignored env TEXTDOMAIN
debug3: Ignored env _
debug3: Ignored env OLDPWD
debug2: channel 0: request shell confirm 1
debug3: send packet: type 98
debug2: channel_input_open_confirmation: channel 0: callback done
debug2: channel 0: open confirm rwindow 0 rmax 32768
debug2: channel 0: rcvd adjust 2097152
debug3: receive packet: type 99
debug2: channel_input_status_confirm: type 99 id 0
debug2: shell request accepted on channel 0
debug2: channel 0: rcvd ext data 227
Hi there, myusername! You've successfully authenticated with the key named myusername@email.com, but Gitea does not provide shell access.
If this is unexpected, please log in with password and setup Gitea under another user.
debug2: channel 0: written 227 to efd 6
debug3: receive packet: type 96
debug2: channel 0: rcvd eof
debug2: channel 0: output open -> drain
debug2: channel 0: obuf empty
debug2: channel 0: chan_shutdown_write (i0 o1 sock -1 wfd 5 efd 6 [write])
debug2: channel 0: output drain -> closed
debug3: receive packet: type 98
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
debug3: receive packet: type 98
debug1: client_input_channel_req: channel 0 rtype eow@openssh.com reply 0
debug2: channel 0: rcvd eow
debug2: channel 0: chan_shutdown_read (i0 o3 sock -1 wfd 4 efd 6 [write])
debug2: channel 0: input open -> closed
debug3: receive packet: type 97
debug2: channel 0: rcvd close
debug3: channel 0: will not send data after close
debug2: channel 0: almost dead
debug2: channel 0: gc: notify user
debug2: channel 0: gc: user detached
debug2: channel 0: send close
debug3: send packet: type 97
debug2: channel 0: is dead
debug2: channel 0: garbage collecting
debug1: channel 0: free: client-session, nchannels 1
debug3: channel 0: status: The following connections are open:
  #0 client-session (t4 r0 i3/0 o3/0 e[write]/0 fd -1/-1/6 sock -1 cc -1)

debug3: send packet: type 1
debug3: fd 1 is not O_NONBLOCK
Transferred: sent 2728, received 3424 bytes, in 0.4 seconds
Bytes per second: sent 7104.6, received 8917.2
debug1: Exit status 0
```

2. SSH session log for Gitea v1.12.3 (session connection failed)
```
...
debug1: Server accepts key: /home/pi/.ssh/gitea-key RSA SHA256:pxkPu96w0CfEom3kXsSGjAUFrgY6pxPYlaOMVB+zBt8 explicit
debug3: sign_and_send_pubkey: RSA SHA256:pxkPu96w0CfEom3kXsSGjAUFrgY6pxPYlaOMVB+zBt8
debug3: sign_and_send_pubkey: signing using rsa-sha2-512
debug3: send packet: type 50
debug3: receive packet: type 52
debug1: Authentication succeeded (publickey).
Authenticated to 192.168.10.30 ([192.168.10.30]:2222).
debug1: channel 0: new [client-session]
debug3: ssh_session2_open: channel_new: 0
debug2: channel 0: send open
debug3: send packet: type 90
debug1: Requesting no-more-sessions@openssh.com
debug3: send packet: type 80
debug1: Entering interactive session.
debug1: pledge: network
debug3: receive packet: type 80
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug3: receive packet: type 4
debug1: Remote: /data/git/.ssh/authorized_keys:2: key options: command user-rc
debug3: receive packet: type 4
debug1: Remote: /data/git/.ssh/authorized_keys:2: key options: command user-rc
debug3: receive packet: type 91
debug2: channel_input_open_confirmation: channel 0: callback start
debug2: fd 3 setting TCP_NODELAY
debug3: ssh_packet_set_tos: set IP_TOS 0x08
debug2: client_session2_setup: id 0
debug1: Sending environment.
debug3: Ignored env SHELL
debug3: Ignored env PWD
debug3: Ignored env LOGNAME
debug3: Ignored env XDG_SESSION_TYPE
debug3: Ignored env HOME
debug1: Sending env LANG = en_GB.UTF-8
debug2: channel 0: request env confirm 0
debug3: send packet: type 98
debug3: Ignored env LS_COLORS
debug3: Ignored env SSH_CONNECTION
debug3: Ignored env XDG_SESSION_CLASS
debug3: Ignored env TERM
debug3: Ignored env USER
debug3: Ignored env SHLVL
debug3: Ignored env XDG_SESSION_ID
debug3: Ignored env XDG_RUNTIME_DIR
debug3: Ignored env SSH_CLIENT
debug3: Ignored env PATH
debug3: Ignored env DBUS_SESSION_BUS_ADDRESS
debug3: Ignored env MAIL
debug3: Ignored env SSH_TTY
debug3: Ignored env TEXTDOMAIN
debug3: Ignored env _
debug3: Ignored env OLDPWD
debug2: channel 0: request shell confirm 1
debug3: send packet: type 98
debug2: channel_input_open_confirmation: channel 0: callback done
debug2: channel 0: open confirm rwindow 0 rmax 32768
debug2: channel 0: rcvd adjust 2097152
debug3: receive packet: type 99
debug2: channel_input_status_confirm: type 99 id 0
debug2: shell request accepted on channel 0
debug3: send packet: type 1
debug1: channel 0: free: client-session, nchannels 1
debug3: channel 0: status: The following connections are open:
  #0 client-session (t4 r0 i0/0 o0/0 e[write]/0 fd 4/5/6 sock -1 cc -1)

debug3: fd 1 is not O_NONBLOCK
Connection to 192.168.10.30 closed by remote host.
Transferred: sent 2708, received 3060 bytes, in 25.5 seconds
Bytes per second: sent 106.0, received 119.8
debug1: Exit status -1
```

Please let me know if you can assist :)

**Comment by Kunde21** [comment]
Thank you for raising the issue.  I'll try to recreate and debug.  

Curious, what board is this running on?  I hope it's not hardware-specific, but I'll want to confirm on the same, if I can.

**Comment by mkenne11** [comment]
Nps thanks for getting back to me. I'm running on a Raspberry pi 4b in containerd (using k3s).
Let me know if you need any further information 😊

**Comment by Kunde21** [comment]
Update:  I'm running out of ideas.   
I've tried k8s deployment, docker container, containerd container, and k3s deployment.  So far, I've been unable to replicate.  

What version of k3s are you running and have you tried updating to latest?

**Comment by mkenne11** [comment]
Thanks for all your work trying to replicate 😊 I'm running the latest k3s (v1.18.6) BTW.
I'll do some more testing on my side to try and debug and find the issue.

**Comment by mkenne11** [comment]
Hi Chad - I tracked down the issue. My Kubernetes deployment template (YAML) set CPU and memory resource limits which weren't high enough to handle the resource requirements for Gitea v1.12 (they have increased since v1.11).
I updated the resource limits and it fixed the issue with SSH pushes hanging. Thanks again :)

**Comment by Kunde21** [comment]
@mkenne11 - Oh, good find.  Can you share the new limits you used?  I'll add that to the docker overview, in case someone else runs into the same one.

**Comment by mkenne11** [comment]
Thanks @Kunde21 - I'm monitoring my cluster with Prometheus which helped identify the issue. I saw a big spike in cpu and memory usage for the gitea pod when I did git push and pulls over SSH.
The resource values I'm using for my gitea pod are:
```
...
resources:
          limits:
            memory: "300Mi"
            cpu: "600m"
          requests:
            memory: "150Mi"
            cpu: "100m"
```

Note, I'm using sqlite as the database and it's just one user using gitea so these are probably the minimal limits on a Raspberry Pi 4.
